### PR TITLE
Upgrade @apollo/client v3.1.3

### LINKED ISSUE #1002: Upgrade to Apollo v3
As a Redwood v1 dependency, we want to upgrade Apollo (client and server) to v3:
https://github.com/apollographql/apollo-client/releases/tag/v3.0.0

Next Steps:
- identify specific breaking changes

### PR DESCRIPTION
**Pull request by thedavidprice**
I'm shocked this is passing my local tests and works just fine running my local Redwood Tutorial Blog project. I've only upgraded apollo packages and swapped `apollo-client` for `@apollo/client`. It's more likely I’ve missed something and upgraded incorrectly 'cause it's never this easy, right? ¯\_(ツ)_/¯

Apollo Client v3 Release Notes and breaking changes:
https://github.com/apollographql/apollo-client/releases/tag/v3.0.0

If it's true this upgrade works without issues, it seems we should still tread lightly putting this into a release as it's likely to break projects (especially those relying on apollo caching):
- maybe look to v0.18 release
- could do an actual release candidate and/or have a canary release that's tested on production projects

What say you @peterp?

Closes #1002

### CODE REVIEW

File: packages/web/package.json [hunk 1]
```diff
@@ -9,16 +9,16 @@
...
-    "apollo-cache": "^1.3.4",
-    "apollo-client": "^2.6.8",
-    "apollo-link": "^1.2.13",
-    "apollo-utilities": "^1.3.3",
-    "core-js": "3.6.4",
+    "apollo-boost": "0.4.9",
+    "apollo-cache": "^1.3.5",
+    "apollo-link": "^1.2.14",
+    "apollo-utilities": "^1.3.4",
+    "core-js": "3.6.5",
```
Comment (peterp) [inline packages/web/package.json:21]: `core-js` should stay the same across all of our packages. I should add the resolutions stuff to the main package to pin it, maybe we can upgrade in a seperate PR and I can show you why.

```suggestion
    "core-js": "3.6.4",
```

File: packages/web/package.json [hunk 2]
```diff
@@ -9,16 +9,16 @@
...
-    "apollo-boost": "0.4.7",
-    "apollo-cache": "^1.3.4",
-    "apollo-client": "^2.6.8",
-    "apollo-link": "^1.2.13",
-    "apollo-utilities": "^1.3.3",
-    "core-js": "3.6.4",
+    "apollo-boost": "0.4.9",
+    "apollo-cache": "^1.3.5",
+    "apollo-link": "^1.2.14",
+    "apollo-utilities": "^1.3.4",
```
Comment (peterp) [inline packages/web/package.json:20]: I don't think these packages are required anymore?

File: packages/web/package.json [hunk 3]
```diff
@@ -9,16 +9,16 @@
   "types": "src/index.d.ts",
   "license": "MIT",
   "dependencies": {
-    "@apollo/react-components": "^3.1.3",
-    "@apollo/react-hooks": "^3.1.3",
+    "@apollo/client": "^3.1.3",
+    "@apollo/react-components": "^4.0.0",
+    "@apollo/react-hooks": "^4.0.0",
```
Comment (peterp) [inline packages/web/package.json:14]: The react-hooks library is now included in apollo-client.

File: packages/web/package.json [hunk 4]
```diff
@@ -9,16 +9,16 @@
   "types": "src/index.d.ts",
   "license": "MIT",
   "dependencies": {
-    "@apollo/react-components": "^3.1.3",
-    "@apollo/react-hooks": "^3.1.3",
+    "@apollo/client": "^3.1.3",
+    "@apollo/react-components": "^4.0.0",
```
Comment (peterp) [inline packages/web/package.json:13]: The react-components are now included in apollo-client.

**Comment by peterp** [comment]
I went through the upgrading guide and it seems like the approach should be to delete all the other packages and only install `@apollo/client`: https://www.apollographql.com/docs/react/migrating/apollo-client-3-migration/

We would need to modify the imports to match this new world.

**Comment by thedavidprice** [comment]
@peterp Got it. And thanks for taking a look here!

This PR was primarily to get the conversation going. I was curious post Roadmap planning and given this super rough first pass optimistic it won't be too bad.

I can take the next pass at the Migration guide. BUT if we end up deleting this PR and opening another, all good.

File: packages/web/package.json (hunk 1 above)
Comment (thedavidprice) [reply packages/web/package.json:21]: Ah, roger that. Understood and didn't intend for this to be included.

**Comment by thedavidprice** [comment]
First pass through the Apollo upgrade documentation replacing imports. Generated a new yarn.lock, which I can revert if it causes problems.

Passing tests but not working in my project. Seems `RedwoodProvider.js` config is the next step to work on.

Also, I noticed `packages/core/src/configs/browser/jest.createConfig.ts` is accessing `@apollo/react-common` on L29-L31, which I don't think will exist anymore. Not sure what this is doing so will need to loop in Robert.

**Comment by peterp** [comment]
@thedavidprice I think this is good to go now, my only real concern is how to communicate the **breaking cache changes**: https://www.apollographql.com/docs/react/migrating/apollo-client-3-migration/#breaking-cache-changes

**Comment by peterp** [comment]
@thedavidprice Setting this to "DRAFT" so that we don't accidentally merge it :)

**Comment by thedavidprice** [comment]
@peterp thanks again for moving this forward! 

The last item I had on my list is this random bit within Jest config:
https://github.com/redwoodjs/redwood/blob/ef48c22a4320e01628bfe83b9bb11bce22fb0602/packages/core/src/configs/browser/jest.createConfig.ts#L28-L31

`@apollo/react-common` has been deprecated and replaced with `@apollo/client/react` based on [this upgrade info](https://www.apollographql.com/docs/react/migrating/apollo-client-3-migration/#react-apollo). I've made the change in the latest commit, but please do double-check as I didn't do the original Jest config setup. I'm a little unclear what's required in the config itself.

**Comment by peterp** [comment]
@thedavidprice Ah, we can remove those.

**Comment by peterp** [comment]
Without Apollo v3:

image

With Apollo v3:

image

200 kbit reduction. I am 100% confident that we do not need "GraphQL" either, since we transpile at build time.

**Comment by peterp** [comment]
With Apollo v3.
image

**Pull request merged**